sdks/python/apache_beam/ml/inference/vertex_ai_inference.py

```diff
@@ -219,109 +217,81 @@ def validate_inference_args(self, inference_args: Optional[dict[str, Any]]):
-            _ = self._retrieve_endpoint(
-                endpoint_id=full_endpoint_id_for_check, 
-                project=self.project_id, 
-                location=self.location,
-                is_private=self.private)
-            LOGGER.info("Initial liveness check successful.")
-        except ValueError as e:
-            LOGGER.warning("Initial liveness check for endpoint %s failed: %s. "
-                           "Will retry in load_model.", self.endpoint_name, e)
-        except Exception as e:
-             LOGGER.warning("Unexpected error during initial liveness check for endpoint %s: %s. "
-                           "Will retry in load_model.", self.endpoint_name, e, exc_info=True)
-        # Configure AdaptiveThrottler and throttling metrics for client-side
-        # throttling behavior.
-        # See https://docs.google.com/document/d/1ePorJGZnLbNCmLD9mR7iFYOdPsyDA1rDnTpYnbdrzSU/edit?usp=sharing
-        # for more details.
-        self.throttled_secs = Metrics.counter(
-        VertexAITritonModelHandler, "cumulativeThrottlingSeconds")
-        self.throttler = AdaptiveThrottler(
-        window_ms=1, bucket_ms=1, overload_ratio=2) 
-    def load_model(self) -> aiplatform.Endpoint:
-        """Loads the Endpoint object for inference, performing liveness check."""
-        
-        endpoint_id_to_load = self.endpoint_name
-        if not endpoint_id_to_load.startswith('projects/'):
-             endpoint_id_to_load = f"projects/{self.project_id}/locations/{self.location}/endpoints/{self.endpoint_name}"
-
-        LOGGER.info("Loading/Retrieving Vertex AI endpoint: %s", endpoint_id_to_load)
-        ep = self._retrieve_endpoint(
-            endpoint_id=endpoint_id_to_load,
-            project=self.project_id,
-            location=self.location,
-            is_private=self.private)
-        LOGGER.info("Successfully retrieved endpoint object: %s", ep.name)
-        return ep
+        if private and network is None:
+            raise ValueError("A VPC network must be provided for a private endpoint.")
+
+        aiplatform.init(
+            project=project,
+            location=location,
+            experiment=experiment,
+            network=network
+        )
+        self.project = project
+        self.endpoint_name = endpoint_id
+        self.location = location
+        self.is_private = private
+
+        self.endpoint = self._retrieve_endpoint(  
+            self.endpoint_name, self.project, self.location, self.is_private
+        )
+
+        super().__init__(
+            namespace='VertexAITritonModelHandler',
+            retry_filter=_retry_on_appropriate_gcp_error,
+            **kwargs
+        )
+    def create_client(self) -> aiplatform.Endpoint:
+        """
+        Create the client for inference.
 
+        Returns:
+            aiplatform.Endpoint object.
+        """
+        return self.endpoint
+    
     def _retrieve_endpoint(
```

Comment:
I cannot find any sort of discussion around public versus private triton endpoints, but as I've said before the aiplatform.Endpoint classes aren't what you should be using anyway.